pkg/ccl/benchccl/rttanalysisccl/rttanalysisccl_test: TestBenchmarkExpectation failed [setting updated but timed out waiting to read new value]

pkg/ccl/benchccl/rttanalysisccl/rttanalysisccl_test.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20523666?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20523666?buildTab=artifacts#/) on release-25.4 @ [2a25514b82b8ce169e60e2526320052da891d2a9](https://github.com/cockroachdb/cockroach/commits/2a25514b82b8ce169e60e2526320052da891d2a9):


```
=== RUN   TestBenchmarkExpectation
    validate_benchmark_data.go:186: [/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/17132/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/benchccl/rttanalysisccl/rttanalysisccl_test_/rttanalysisccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/benchccl/rttanalysisccl/rttanalysisccl_test_/rttanalysisccl_test --test.v=test2json --test.bench=^$]
I250926 03:33:18.412644 1 (gostd) rand.go:268  [-] 1  random seed: -2916879196197594892
=== RUN   TestBenchmarkExpectation
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/7418df5abfb13dbfaf6c0beee25f240a/logTestBenchmarkExpectation2287784375
    test_log_scope.go:82: use -show-logs to present logs inline
    validate_benchmark_data.go:120: success: expected AlterSurvivalGoals/alter_empty_database_from_region_to_zone to perform KV lookups in [12, 12], got 12
    validate_benchmark_data.go:120: success: expected AlterTableLocality/alter_from_regional_by_table_to_global to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected AlterTableLocality/alter_from_regional_by_table_to_rbr to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:120: success: expected AlterTableLocality/alter_from_rbr_to_global to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:120: success: expected AlterTableLocality/alter_from_global_to_rbr to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:120: success: expected AlterTableLocality/alter_from_rbr_to_regional_by_table to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:120: success: expected AlterPrimaryRegion/alter_populated_database_alter_primary_region to perform KV lookups in [12, 14], got 12
    validate_benchmark_data.go:120: success: expected AlterSurvivalGoals/alter_populated_database_from_region_to_zone to perform KV lookups in [22, 22], got 22
    validate_benchmark_data.go:120: success: expected AlterSurvivalGoals/alter_empty_database_from_zone_to_region to perform KV lookups in [12, 12], got 12
    validate_benchmark_data.go:120: success: expected AlterPrimaryRegion/alter_empty_database_set_initial_primary_region to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected AlterPrimaryRegion/alter_empty_database_alter_primary_region to perform KV lookups in [12, 14], got 12
    validate_benchmark_data.go:120: success: expected AlterSurvivalGoals/alter_populated_database_from_zone_to_region to perform KV lookups in [22, 22], got 22
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/select_from_crdb_internal.zones_(10_tables) to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:120: success: expected AlterTableLocality/alter_from_global_to_regional_by_table to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected AlterPrimaryRegion/alter_populated_database_set_initial_primary_region to perform KV lookups in [14, 14], got 14
    validate_benchmark_data.go:120: success: expected AlterRegions/alter_populated_database_add_region to perform KV lookups in [11, 11], got 11
    validate_benchmark_data.go:120: success: expected AlterRegions/alter_empty_database_drop_region to perform KV lookups in [12, 16], got 14
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/select_from_crdb_internal.zones_(100_tables) to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:120: success: expected AlterRegions/alter_populated_database_drop_region to perform KV lookups in [14, 14], got 14
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/select_from_crdb_internal.zones_(50_tables) to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:73: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:74: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:107: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7418df5abfb13dbfaf6c0beee25f240a/logTestBenchmarkExpectation2287784375
--- FAIL: TestBenchmarkExpectation (95.21s)
--- FAIL: TestBenchmarkExpectation (95.80s)
=== RUN   TestBenchmarkExpectation/AlterRegions
--- FAIL: TestBenchmarkExpectation/AlterRegions (92.42s)
=== RUN   TestBenchmarkExpectation/AlterRegions/alter_empty_database_add_region
    multi_region_bench_test.go:37: pq: setting updated but timed out waiting to read new value
--- FAIL: TestBenchmarkExpectation/AlterRegions/alter_empty_database_add_region (84.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54813

--- a/pkg/sql/set_cluster_setting.go
+++ b/pkg/sql/set_cluster_setting.go
@@ -40,6 +40,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sessioninit"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltelemetry"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/buildutil"
 	"github.com/cockroachdb/cockroach/pkg/util/humanizeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/log/eventpb"
@@ -701,7 +702,12 @@ func waitForSettingUpdate(
 	}
 	errNotReady := errors.New("setting updated but timed out waiting to read new value")
 	var observed string
-	err := retry.ForDuration(10*time.Second, func() error {
+	defaultDuration := 10 * time.Second
+	// Bench tests maybe more prone to flaking, so use a longer time limit for them.
+	if buildutil.CrdbBenchBuild {
+		defaultDuration *= 3
+	}
+	err := retry.ForDuration(defaultDuration, func() error {
 		observed = setting.Encoded(&execCfg.Settings.SV)
 		if observed != expectedEncodedValue {
 			return errNotReady

